dockstore-integration-testing/src/test/java/io/dockstore/client/cli/WorkflowIT.java

```diff
@@ -291,6 +292,42 @@ public void testCWLLanguageParsingInformation() {
         Assert.assertTrue(parsedInformation.isHasHTTPImports());
     }
 
+    /**
+     * Test the precedence of CWL requirements and hints.
+     * The workflow steps in the referenced workflow are named to indicate whether the step and invoked command
+     * defined requirements or hints.  For example, the workflow step id "requirement_none" means that the workflow
+     * step defined a requirement, and the invoked command did not define a requirement or a hint.
+     *
+     * <p> According to the CWL spec, a child inherits requirements and hints from its parent.  Child requirements
+     * override parent requirements, child hints override parent hints, and if both are present at a node,
+     * requirements take precedence over hints.
+     */
+    @Test
+    public void testCWLRequirementsAndHints() {
+        final ApiClient webClient = getWebClient(USER_2_USERNAME, testingPostgres);
+        WorkflowsApi workflowApi = new WorkflowsApi(webClient);
+        Workflow cwlWorkflow = workflowApi
+               .manualRegister(SourceControl.GITHUB.name(), "dockstore-testing/cwl-requirements-and-hints", "/parent.cwl", "CWL",
+                       DescriptorLanguage.CWL.toString(), "/parent.json");
+        Long cwlId = cwlWorkflow.getId();
+        workflowApi.refresh(cwlId, false);
+        Workflow workflow = workflowApi.getWorkflow(cwlId, null);
+        WorkflowVersion version = workflow.getWorkflowVersions().stream().filter(v -> v.getName().equals("main")).findFirst().get();
```

Comment:
1. You could create a tag, just in case somebody some day decides, unlikely, to tweak the main branch.
2. If you did a unit test instead, you'd remove the dependency on an external repo, it would run faster. OTOH, you're make sure the version gets created, so...

Reply:
True, I'd forgotten that the tool content can be retrieved via `getContent`.  Integration test converted to unit test.